### PR TITLE
[Enhance] modify ppyoloe regmax as yolov8

### PR DESCRIPTION
**Pull request by triple-Mu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by hhaAndroid — APPROVED** [review]

**Comment by Nioolek** [comment]
v8和ppyoloe虽然reg_max参数都是80，但其实在网络里一个是输出81，一个是输出80，是有区别的，只改head应该会导致出错吧？有跑test进行测试吗？

**Comment by Nioolek** [comment]
test能对齐精度吗？

File: mmyolo/models/dense_heads/ppyoloe_head.py [hunk 1]
```diff
@@ -350,11 +355,11 @@ def loss_by_feat(
...
-                [1, 1, (self.head_module.reg_max + 1) * 4])
+                [1, 1, self.head_module.reg_max * 4])
 
             pred_dist_pos = torch.masked_select(
                 flatten_dist_preds,
-                dist_mask).reshape([-1, 4, self.head_module.reg_max + 1])
+                dist_mask).reshape([-1, 4, self.head_module.reg_max])
             assigned_ltrb = self.bbox_coder.encode(
                 self.flatten_priors_train[..., :2] / self.stride_tensor,
                 assigned_bboxes,
```
Comment (hhaAndroid) [inline mmyolo/models/dense_heads/ppyoloe_head.py:365]: assigned_ltrb = self.bbox_coder.encode() 里面有 max_dis 好像没有改

File: mmyolo/models/dense_heads/ppyoloe_head.py [hunk 2]
```diff
@@ -350,11 +355,11 @@ def loss_by_feat(
...
             # dfl loss
             dist_mask = fg_mask_pre_prior.unsqueeze(-1).repeat(
-                [1, 1, (self.head_module.reg_max + 1) * 4])
+                [1, 1, self.head_module.reg_max * 4])
 
             pred_dist_pos = torch.masked_select(
                 flatten_dist_preds,
-                dist_mask).reshape([-1, 4, self.head_module.reg_max + 1])
+                dist_mask).reshape([-1, 4, self.head_module.reg_max])
             assigned_ltrb = self.bbox_coder.encode(
```
Comment (hhaAndroid) [inline mmyolo/models/dense_heads/ppyoloe_head.py:363]: assigned_ltrb = self.bbox_coder.encode() 里面有 max_dis 好像没有改

Comment (triple-Mu) [reply mmyolo/models/dense_heads/ppyoloe_head.py:363]: 这个max_dis在哪里呀

File: mmyolo/models/dense_heads/ppyoloe_head.py (hunk 1 above)
Comment (hhaAndroid) [reply mmyolo/models/dense_heads/ppyoloe_head.py:365]: assigned_ltrb = self.bbox_coder.encode(
                self.flatten_priors_train[..., :2] / self.stride_tensor,
                assigned_bboxes,
                max_dis=self.head_module.reg_max,
                eps=0.01)

File: mmyolo/models/dense_heads/ppyoloe_head.py (hunk 2 above)
Comment (hhaAndroid) [reply mmyolo/models/dense_heads/ppyoloe_head.py:363]: assigned_ltrb = self.bbox_coder.encode(
            self.flatten_priors_train[..., :2] / self.stride_tensor,
            assigned_bboxes,
            max_dis=self.head_module.reg_max,
            eps=0.01)

**Pull request closed without merging**